### PR TITLE
Dynamic log levels

### PR DESCRIPTION
**Pull request by atye**
# Description

This PR adds log level configuration with the ability to update the log level on the fly.

# Issues

List the issues impacted by this PR:

| Issue ID |
| -------- |
|    #118       |

# Checklist:

- [x] I have performed a self-review of my own changes.

### CODE REVIEW

**Comment by ioeicicdsvc** [comment]
@atye: Thank you for submitting your pull request!

Please make sure you have followed our contributing guidelines. We will review it as soon as possible.

------------------------------------------------------------------------------------------------------------------------
We want your feedback! If you have any questions or suggestions regarding our contributing process/workflow, please reach out to us at karavi@dell.com.

File: cmd/tenant-service/main.go [hunk 1]
```diff
@@ -98,11 +106,21 @@ func main() {
...
 	updateLoggingSettings(log)
 
-	cfgViper.WatchConfig()
-	cfgViper.OnConfigChange(func(e fsnotify.Event) {
+	csmViper.WatchConfig()
+	csmViper.OnConfigChange(func(e fsnotify.Event) {
 		updateLoggingSettings(log)
 	})
 
+	ticker := time.NewTicker(time.Second)
```
Comment (ameer-jabbar) [inline cmd/tenant-service/main.go:114]: Same as my previous comment -- should this code block be removed?

File: cmd/proxy-server/main.go [hunk 2]
```diff
@@ -155,11 +164,21 @@ func run(log *logrus.Entry) error {
...
 	updateLoggingSettings(log)
 
-	cfgViper.WatchConfig()
-	cfgViper.OnConfigChange(func(e fsnotify.Event) {
+	csmViper.WatchConfig()
+	csmViper.OnConfigChange(func(e fsnotify.Event) {
 		updateLoggingSettings(log)
 	})
 
+	ticker := time.NewTicker(time.Second)
```
[the remark this answers is not in the document]
Comment (atye) [reply cmd/proxy-server/main.go]: Removed.

File: cmd/tenant-service/main.go (hunk 1 above)
Comment (atye) [reply cmd/tenant-service/main.go]: Removed.

File: cmd/karavictl/cmd/inject.go [hunk 3]
```diff
@@ -880,6 +880,12 @@ func (lc *ListChangeForMultiArray) injectIntoDeployment(imageAddr, proxyHost str
 
 	// Add a new proxy container...
 	proxyContainer := buildProxyContainer(deploy.Namespace, lc.InjectResources.Secret, imageAddr, proxyHost, insecure)
+	proxyContainer.VolumeMounts = append(proxyContainer.VolumeMounts, corev1.VolumeMount{
+		MountPath: "/etc/karavi-authorization",
+		Name:      "vxflexos-config-params", // {{ .Release.Name }}-config-params
+	})
+	proxyContainer.Args = append(proxyContainer.Args, "--driver-config-params=/etc/karavi-authorization/driver-config-params.yaml")
+	//addDriverConfigMapVolumeMount("driver-config", proxyContainer)
```
Comment (hoppea2) [inline cmd/karavictl/cmd/inject.go:888]: remove this comment

File: cmd/sidecar-proxy/main.go [hunk 4]
```diff
@@ -204,6 +212,40 @@ func run(log *logrus.Entry) error {
 		insecureProxy = true
 	}
 
+	driverConfigParamsFile = flag.String("driver-config-params", "", "Full path to the YAMl file containing the driver ConfigMap")
```
Comment (hoppea2) [inline cmd/sidecar-proxy/main.go:215]: YAML

File: docs/GETTING_STARTED_GUIDE.md [hunk 5]
```diff
@@ -95,6 +95,16 @@ rpm -ivh <rpm_file_name>
...
 
+## Other Configuration Settings
+
+Some settings are not stored in the karavi-config-secret but in the csm-config-params ConfigMap, such as LOG_LEVEL and LOG_FORMAT. To update the karavi-authorization logging settings during runtime, run the below command on the K3s cluster, make your changes, and save the updated configmap data.
+
+```
+k3s kubectl -n karavi edit configmap/csm-config-params
+```
+
+This edit will not update the logging level for the sidecar-proxy containers running in the csi-driver pods. To update the sidecar-proxy logging levels, you must update the associated csi-driver ConfigMap in a similar fashion.
```
Comment (hoppea2) [inline docs/GETTING_STARTED_GUIDE.md:106]: Let's give details on how to update the log levels instead

File: cmd/sidecar-proxy/main.go (hunk 4 above)
Comment (atye) [reply cmd/sidecar-proxy/main.go]: Fixed.

File: cmd/karavictl/cmd/inject.go (hunk 3 above)
Comment (atye) [reply cmd/karavictl/cmd/inject.go]: Fixed.

File: docs/GETTING_STARTED_GUIDE.md (hunk 5 above)
Comment (atye) [reply docs/GETTING_STARTED_GUIDE.md]: Updated with a short blurb using PowerFlex as an example.

**Review by alikdell — APPROVED** [review]

**Review by LJ-Software — APPROVED** [review]

**Review by hoppea2 — APPROVED** [review]

**Pull request merged**